components/masking_functions/src/masking_functions/registration_routines.cpp

```diff
@@ -0,0 +1,1156 @@
+  virtual std::string_view default_ascii_masking_char() const override {
+    return star_ascii_masking_char;
+  }
+  virtual masking_functions::charset_string process(
+      const masking_functions::charset_string &cs_str,
+      const masking_functions::charset_string &masking_char) const override {
+    return masking_functions::mask_inner_alphanum(cs_str, 2, 0, masking_char);
+  }
+};
+
+//
+// mask_uuid(string, [char])
+//
+class mask_uuid_impl final : private mask_impl_base {
+ public:
+  using mask_impl_base::calculate;
+  using mask_impl_base::mask_impl_base;
+
+ private:
+  virtual std::size_t min_length() const override { return 36; }
+  virtual std::size_t max_length() const override { return 36; }
+  virtual std::string_view default_ascii_masking_char() const override {
+    return star_ascii_masking_char;
+  }
+  virtual masking_functions::charset_string process(
+      const masking_functions::charset_string &cs_str,
+      const masking_functions::charset_string &masking_char) const override {
+    auto sresult =
+        masking_functions::mask_inner(cs_str, 0, 36 - 8, masking_char);
+    sresult =
+        masking_functions::mask_inner(sresult, 9, 36 - 9 - 4, masking_char);
+    sresult = masking_functions::mask_inner(sresult, 9 + 5, 36 - 9 - 5 - 4,
+                                            masking_char);
+    sresult = masking_functions::mask_inner(sresult, 9 + 5 + 5,
+                                            36 - 9 - 5 - 5 - 4, masking_char);
+    sresult =
+        masking_functions::mask_inner(sresult, 9 + 5 + 5 + 5, 0, masking_char);
+
+    return sresult;
+  }
+};
+
+//
+// gen_blocklist(string, string, string)
+//
+class gen_blocklist_impl {
+ public:
+  gen_blocklist_impl(mysqlpp::udf_context &ctx) {
+    if (ctx.get_number_of_args() != 3)
+      throw std::invalid_argument{
+          "Wrong argument list: gen_blocklist(string, string, string)"};
+
+    ctx.mark_result_nullable(true);
+    ctx.mark_result_const(false);
+
+    // arg1 - dictionary
+    ctx.mark_arg_nullable(0, false);
+    ctx.set_arg_type(0, STRING_RESULT);
+
+    ctx.mark_arg_nullable(1, false);
+    ctx.set_arg_type(1, STRING_RESULT);
+
+    ctx.mark_arg_nullable(2, false);
+    ctx.set_arg_type(2, STRING_RESULT);
+
+    mysqlpp::udf_context_charset_extension charset_ext{
+        mysql_service_mysql_udf_metadata};
+    set_return_value_collation_from_arg(charset_ext, ctx, 0);
+  }
+
+  mysqlpp::udf_result_t<STRING_RESULT> calculate(
+      const mysqlpp::udf_context &ctx) {
+    const auto cs_term = extract_charset_string_from_arg(ctx, 0);
+    const auto cs_dict_a = extract_charset_string_from_arg(ctx, 1);
+    const auto cs_dict_b = extract_charset_string_from_arg(ctx, 2);
+
+    {
+      masking_functions::sql_context sql_ctx{
+          global_command_services::instance()};
+
+      auto query =
+          global_query_builder::instance()
+              .select_random_term_for_dictionary_and_term(cs_dict_a, cs_term);
```

Comment:
Agree. I was mostly concentrating on refactoring rather than on logic :)
Functions reworked and renamed. Now we have
```
select_random_term_for_dictionary()
check_term_presence_in_dictionary()
```
and internal
```
select_term_for_dictionary_internal(
```